### PR TITLE
Add more granular user rights

### PR DESCRIPTION
**Pull request by Jim3Things**
This change adds granular user rights.  It is a change to the schema, so user entries are automatically updated on use.  As the clients are now part of the same project, the clients only support the new schema.

I've added some unit tests in the simulation project, otherwise the changes are only the minimum to use the new location for can-manage-accounts.  More integration in the UI will happen later, as will more extensive client unit tests.

### CODE REVIEW

File: simulation/internal/services/frontend/DBUsers.go [hunk 1]
```diff
@@ -227,14 +258,16 @@ func (m *DBUsers) UpdatePassword(ctx context.Context, name string, hash []byte,
...
+
 	// Update the entry, retaining all fields save the password hash
 	//
 	user := &pb.User{
 		Name:              old.GetName(),
 		PasswordHash:      hash,
 		UserId:            old.GetUserId(),
 		Enabled:           old.GetEnabled(),
-		CanManageAccounts: old.GetCanManageAccounts(),
+		Rights: old.GetRights(),
```
Comment (MikeCJohnsonUk) [inline simulation/internal/services/frontend/DBUsers.go:270]: Alignment?

File: simulation/internal/services/frontend/users.go [hunk 2]
```diff
@@ -212,7 +223,7 @@ func handlerUserRead(w http.ResponseWriter, r *http.Request) {
 
 	ext := &pb.UserPublic{
 		Enabled:           u.Enabled,
-		CanManageAccounts: u.CanManageAccounts,
+		Rights: u.Rights,
```
Comment (MikeCJohnsonUk) [inline simulation/internal/services/frontend/users.go:226]: Alignment?

File: simulation/internal/services/frontend/users.go [hunk 3]
```diff
@@ -304,7 +317,7 @@ func handlerUserUpdate(w http.ResponseWriter, r *http.Request) {
 
 	ext := &pb.UserPublic{
 		Enabled:           newVer.Enabled,
-		CanManageAccounts: newVer.CanManageAccounts,
+		Rights: newVer.Rights,
```
Comment (MikeCJohnsonUk) [inline simulation/internal/services/frontend/users.go:320]: Alignment?

File: simulation/internal/services/frontend/users.go [hunk 4]
```diff
@@ -569,7 +582,7 @@ func userAdd(
 		Name:              name,
 		PasswordHash:      passwordHash,
 		Enabled:           enabled,
-		CanManageAccounts: accountManager,
+		Rights: rights,
```
Comment (MikeCJohnsonUk) [inline simulation/internal/services/frontend/users.go:585]: Alignment?

File: simulation/internal/clients/store/storeapi_test.go [hunk 5]
```diff
@@ -244,7 +244,7 @@ func TestUpdate(t *testing.T) {
 		PasswordHash:      passwordHash,
 		UserId:            1,
 		Enabled:           true,
-		CanManageAccounts: false,
+		Rights: &pb.Rights{CanManageAccounts: true},
```
[the remark this answers is not in the document]
Comment (Jim3Things) [reply simulation/internal/clients/store/storeapi_test.go:247]: No, missed this.  Doesn't change the test outcome, but I've flipped it to false.

File: simulation/internal/services/frontend/DBUsers.go [hunk 6]
```diff
@@ -187,7 +216,7 @@ func (m *DBUsers) Update(ctx context.Context, name string, u *pb.UserUpdate, mat
 		PasswordHash:      old.GetPasswordHash(),
 		UserId:            old.GetUserId(),
 		Enabled:           u.GetEnabled(),
-		CanManageAccounts: u.GetCanManageAccounts(),
+		Rights: u.GetRights(),
```
[the remark this answers is not in the document]
Comment (Jim3Things) [reply simulation/internal/services/frontend/DBUsers.go:219]: fixed all these.

**Review by MikeCJohnsonUk — APPROVED** [review]

**Pull request merged**